File: src/models/fullTime.ts
import { Employee } from "./employee";
import { IFullTimeEmployee } from "../interfaces/iFullTime";

export class fullTime extends Employee implements IFullTimeEmployee {
    constructor(name: string, id: number, public annualSalary: number) {
        super(name, id);
    }

    calculateSalary(): number {
        return this.annualSalary;
    }
}

File: src/models/fullTime.cpp
#include "FullTimeEmployee.h"
#include <iostream>
using namespace std;

FullTimeEmployee::FullTimeEmployee(const string& empName, int empId, float salary)
    : Employee(empName, empId), annualSalary(salary) {}

float FullTimeEmployee::calculateSalary() const {
    return annualSalary;
}